youtube_dl/extractor/brightcove.py

```diff
@@ -501,22 +504,39 @@ def _extract_urls(webpage):
                 r'<iframe[^>]+src=(["\'])((?:https?:)?//players\.brightcove\.net/\d+/[^/]+/index\.html.+?)\1', webpage):
             entries.append(url if url.startswith('http') else 'http:' + url)
 
-        # Look for embed_in_page embeds [2]
-        for video_id, account_id, player_id, embed in re.findall(
-                # According to examples from [3] it's unclear whether video id
-                # may be optional and what to do when it is
-                # According to [4] data-video-id may be prefixed with ref:
-                r'''(?sx)
-                    <video[^>]+
-                        data-video-id=["\'](\d+|ref:[^"\']+)["\'][^>]*>.*?
-                    </video>.*?
-                    <script[^>]+
-                        src=["\'](?:https?:)?//players\.brightcove\.net/
-                        (\d+)/([^/]+)_([^/]+)/index(?:\.min)?\.js
-                ''', webpage):
-            entries.append(
-                'http://players.brightcove.net/%s/%s_%s/index.html?videoId=%s'
-                % (account_id, player_id, embed, video_id))
+        # Look for <video> tags [2] and embed_in_page embeds [3]
+        # [2] looks like:
+        # <video data-video-id="5320421710001" data-account="245991542" data-player="SJWAiyYWg" data-embed="default" class="video-js" controls itemscope itemtype="http://schema.org/VideoObject">
+
+        for video, script_tag, account_id, player_id, embed in re.findall(
+            r'''(?isx)
+                (<video[^>]+>)
+                (?:.*?
+                  (<script[^>]+
+                    src=["\'](?:https?:)?//players\.brightcove\.net/
+                    (\d+)/([^/]+)_([^/]+)/index(?:\.min)?\.js
+                  )
+                )?
```

Comment:
> one possible solution is to separate brightcove video tag only extraction (without brightcove indication) to another ambiguous method in BrightcoveNewIE and call it very last in generic extractor as a last resort. 

This is similar to what I was suggesting. Your suggestion is simpler but less general.
I guess the question is really what we can expect to see in the future from other embedded players that use `<video>` tags like this. Should we plan ahead and produce a scheme for them that will work reasonably (my suggestion), or should we only handle Brightcove today and worry about generalizing it later (yours).

I'm not sure what you mean by _ambiguous_, though.

One consequence of this separation is that it basically undoes the merging that @Yan12125 had suggested I implement in https://github.com/rg3/youtube-dl/pull/12099#discussion_r101892750.
It may be that we are letting "best" be the enemy of "good enough."

Ah well…

> Another solution is to prohibit such generic embeds at all and add a separate extractor for each such case.

This doesn't seem good -- the examples we see here are all about Brightcove, and trying to add them site by site is basically maintaining a list of websites that do this and that list will always be incomplete and annoying to maintain.